packages/react-instantsearch-dom/src/components/SearchBox.js

```diff
@@ -115,19 +115,13 @@ class SearchBox extends Component {
     document.removeEventListener('keydown', this.onKeyDown);
   }
 
-  componentWillReceiveProps(nextProps) {
-    // @TODO: should component maybe be controlled, otherwise Derived
-
-    // Reset query when the searchParameters query has changed.
-    // This is kind of an anti-pattern (props in state), but it works here
-    // since we know for sure that searchParameters having changed means a
-    // new search has been triggered.
+  componentDidUpdate(prevProps) {
```

Comment:
I think this should be `getDerivedStateFromProps`, so it happens before render

Reply:
(especially since we are settings state exclusively inside it)

Reply:
According to the unit tests and the manual tests on the Storybook, it works as previously. There might be more renders though. I need to check that more thoroughly.

Reply:
might be the case that it's batched, but not sure how to count that

Reply:
It's now rendered twice when hitting <kbd>Enter</kbd> in `{ searchAsYouType: false }` mode. I added more info about it in the PR description.

Reply:
>You may call setState() immediately in componentDidUpdate() but note that it must be wrapped in a condition like in the example above, or you’ll cause an infinite loop. It would also cause an extra re-rendering which, while not visible to the user, can affect the component performance. If you’re trying to “mirror” some state to a prop coming from above, consider using the prop directly instead. Read more about why copying props into state causes bugs.

https://reactjs.org/docs/react-component.html#componentdidupdate